### PR TITLE
[WIP] - Rose Packaging PR

### PR DESCRIPTION
**Pull request by wxtim**
# Make rose a PyPi package.
## Steps
- [x] Convert Rose into a PyPi package and get the local tests running.
- [ ] Convert Rosie into a PyPi package (this will contain a dependency on rose).
- [ ] Replace the bin directory with wrappers scripts created by the setup file.
- [ ] Make the tests run on Travis.

## Problems
Currently waiting on fixes to Rosie on master: PR #2355 before rebase.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Turns out we can auto-upgrade `import rose` to `import metomi.rose` in a few lines piggybacking on the `2to3` command.

I've stuck up a repo which may be of assistance to people upgrading Rose Macros etc. As and when we make changes just stick a new transformer in here:

https://github.com/oliver-sanders/rose2to3

File: bin/rose-app-run [hunk 1]
```diff
@@ -18,10 +18,10 @@
 # along with Rose. If not, see <http://www.gnu.org/licenses/>.
 #-------------------------------------------------------------------------------
 # NAME
-#     rose app-run
+#     metomi.rose app-run
```
[the remark this answers is not in the document]
Comment (wxtim) [reply bin/rose-app-run]: Ay!

File: bin/rose-config-diff [hunk 2]
```diff
@@ -90,28 +90,28 @@
 #           diff-tool=diff3
 #           gdiff-tool=kompare
 #
-#        in your site or user Rose configuration (`rose.conf`).
+#        in your site or user Rose configuration (`metomi.rose.conf`).
 #
-#     [rose-config-diff]properties, [rose-config-diff]ignore{...}
+#     [metomi.rose-config-diff]properties, [rose-config-diff]ignore{...}
```
[the remark this answers is not in the document]
Comment (wxtim) [reply bin/rose-config-diff]: Oc.

**Comment by wxtim** [comment]
Closed in favour of #2360

**Pull request closed without merging**